mods/ra/rules/structures.yaml

```diff
@@ -520,6 +520,9 @@ DOME:
 	InfiltrateForExploration:
 	DetectCloaked:
 		Range: 10c0
+		UpgradeTypes: powered
+		UpgradeMaxEnabledLevel: 0
+		UpgradeMinEnabledLevel: -1
```

Comment:
Upgrade types are additive, and you cannot revoke upgrades granted with one trait by another trait.  Negative upgrade levels can't happen.